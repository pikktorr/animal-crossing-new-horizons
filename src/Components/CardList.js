import React from "react";
import Card from "./Card";

const CardList = ({ villagers }) => {
  return (
    <div>
      {villagers.map((villager) => {
        console.log(villager.name["name-EUen"]);
        return (
          <Card
            key={villager.id}
            id={villager.id}
            name={villager.name["name-EUen"]}
            phrase={villager["catch-phrase"]}
          />
        );
      })}
    </div>
  );
};

export default CardList;
