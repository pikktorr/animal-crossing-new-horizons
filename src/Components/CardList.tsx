import React from 'react';
import Card from './Card';
import { Villager } from '../App/App';

type CardListProps = {
  villagers: Villager[];
};

const CardList = ({ villagers }: CardListProps) => {
  return (
    <div className="flex flex-wrap gap-4 justify-center m-10">
      {villagers.map((villager) => {
        return (
          <Card
            key={villager.id}
            id={villager.id}
            name={villager.name['name-USen']}
            phrase={villager['catch-phrase']}
          />
        );
      })}
    </div>
  );
};

export default CardList;
