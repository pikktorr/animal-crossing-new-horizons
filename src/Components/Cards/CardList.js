import React from "react";
import Card from "./Card";

const CardList = ({ villagers }) => {
  return (
    <div>
      {Object.entries(villagers).map((villager) => {
        const animalObject = villager[1];
        return (
          <Card
            key={animalObject.id}
            id={animalObject.id}
            name={animalObject.name["name-en"]}
            phrase={animalObject["catch-phrase"]}
          />
        );
      })}
    </div>
  );
};

export default CardList;
