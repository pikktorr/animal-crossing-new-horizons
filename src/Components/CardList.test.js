import React from "react";
import CardList from "./CardList";
import { shallow } from "enzyme";

test("expect to render CardList component", () => {
  const mockVillagers = [
    {
      key: 1,
      id: 1,
      name: { "name-EUen": "Jester" },
      "catch-phrase": "Micki-Mock",
    },
  ];
  expect(
    shallow(<CardList villagers={mockVillagers} />).debug()
  ).toMatchSnapshot();
});
