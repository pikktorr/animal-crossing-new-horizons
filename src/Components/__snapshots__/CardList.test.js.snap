// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
"<div>
  <Card id={1} name=\\"Jester\\" phrase=\\"Micki-Mock\\" />
</div>"
`;
